### PR TITLE
[Merged by Bors] - feat: port Order.GameAdd

### PR DESCRIPTION
**Pull request by dwarn**
mathlib SHA: fd47bdf09e90f553519c712378e651975fe8c829

### CODE REVIEW

File: Mathlib/Order/GameAdd.lean [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  decreases at a time.
+
+## Todo
+
+- Add custom `induction` and `fix` lemmas.
+- Define `Sym2.GameAdd`.
+-/
+
+
+variable {α β : Type _} (rα : α → α → Prop) (rβ : β → β → Prop)
```
Comment (mcdoll) [inline Mathlib/Order/GameAdd.lean:30]:
```suggestion
variable (rα : α → α → Prop) (rβ : β → β → Prop)
```
see https://github.com/leanprover-community/mathlib4/wiki#auto-implicits

File: Mathlib/Order/GameAdd.lean [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+theorem rprod_le_transGen_gameAdd : Prod.RProd rα rβ ≤ Relation.TransGen (GameAdd rα rβ)
+  | _, _, h => h.rec (by
+      intro _ _ _ _ hα hβ
+      exact Relation.TransGen.tail (Relation.TransGen.single $ GameAdd.fst hα) (GameAdd.snd hβ))
+#align prod.rprod_le_trans_gen_game_add Prod.rprod_le_transGen_gameAdd
+
+end Prod
+
+variable {rα rβ}
+
```
Comment (mcdoll) [inline Mathlib/Order/GameAdd.lean:59]:
```suggestion
```

File: Mathlib/Order/GameAdd.lean [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+/-- If `a` is accessible under `rα` and `b` is accessible under `rβ`, then `(a, b)` is
+  accessible under `Prod.GameAdd rα rβ`. Notice that `Prod.lexAccessible` requires the
+  stronger condition `∀ b, Acc rβ b`. -/
+theorem Acc.prod_gameAdd {a b} (ha : Acc rα a) (hb : Acc rβ b) :
```
Comment (mcdoll) [inline Mathlib/Order/GameAdd.lean:63]:
```suggestion
theorem Acc.prod_gameAdd (ha : Acc rα a) (hb : Acc rβ b) :
```

**Review by mcdoll — APPROVED** [review]
LGTM

I like autoimplicits here, because `α` and `β` are almost always just arbitrary types.

**Comment by semorrison** [comment]
After making the changes, Moritz suggested,

bors d+

**Pull request closed without merging**